src/lib.rs

```diff
@@ -201,6 +202,13 @@ impl Framebuffer {
             .unwrap();
     }
 
+    ///Reads a frame from the Framebuffer.
+    pub fn read_frame(&self, frame: &mut Vec<u8>) {
```

Comment:
Using &mut Vec<u8> as parameter requires the user of this function to use a Vec, which will allocate the required memory to fit the entire frame. This means that after every read the user will have to call .clear() on the vector before being able to use the same vector for the next read. Also it would not be possible to only read a small part of the frame.

Wouldn't it be better to change it to:
```rust
pub fn read_frame(&self, frame: &mut [u8]) {
```
and then use `.read_exact(frame)` later on?

Reply:
For reading parts of the framebuffer I would suggest we implement `Index<Range>` so that people can read it any way they want, e.g. `let v = fb[4096..8192].to_vec();`.

Reply:
For reading the entire frame I wanted to use Vec to not have to deal with the size(preallocating memory in case of an array)... but I guess we could also solve this in terms of `Index<Range>` => `let v = fb[..].to_vec()`, `let slice = fb[..]`, and maybe writing too using `IndexMut`.

Reply:
This would give the user the choice to do whatever they want with the returned slice